mempool/cat/cache.go

```diff
@@ -95,6 +95,81 @@ func (c *LRUTxCache) Has(txKey types.TxKey) bool {
 	return ok
 }
 
+// cacheEntry stores both the transaction key and error code
+type cacheEntry struct {
+	key  types.TxKey
+	code uint32
+}
+
+// RejectedTxCache is a cache of rejected transactions. It wraps LRUTxCache
+// to store the error code for a transaction that has been rejected.
+type RejectedTxCache struct {
+	cache *LRUTxCache
+}
+
+// NewRejectedTxCache creates a new rejected tx cache.
+func NewRejectedTxCache(cacheSize int) *RejectedTxCache {
+	return &RejectedTxCache{
+		cache: NewLRUTxCache(cacheSize),
```

Comment:
when does this new cache get pruned?

Reply:
in `flush()` https://github.com/celestiaorg/celestia-core/blob/61cdad4aadcfb5b96243c1ce8cbe10292224c4a2/mempool/cat/pool.go#L376C7-L376C22

also iirc we don't do the same for priority mempool